### PR TITLE
Add pytest-json package

### PR DESCRIPTION
**Pull request by goanpeca**
@mattcl would you be interested in being a maintainer of this conda recipe?

You can read more about conda-forge [here](https://conda-forge.github.io/)

### CODE REVIEW

File: recipes/pytest-json/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{% set version = "0.4.0" %}
+{% set hash_type = "sha256" %}
+{% set hash = "8bf4e1be1691f4416bc12b14785b5ad9e842887b0b2b2d61b37dcb555b208630" %}
+
+package:
+  name: pydocstyle
```
Comment (nicoddemus) [inline recipes/pytest-json/meta.yaml]: Shouldn't be `pytest-json`?

File: recipes/pytest-json/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    - python
+
+test:
+  imports:
+    - pytest_json
+
+about:
+  home: https://github.com/mattcl/pytest-json
+  license: MIT License
+  license_file: LICENSE
```
Comment (nicoddemus) [inline recipes/pytest-json/meta.yaml]: It seems this file is in the repository, but missing from the package. 😬

Comment (goanpeca) [inline recipes/pytest-json/meta.yaml]: Will fix

**Comment by goanpeca** [comment]
@jakirkham build have not progressed in like 12 hours... is this normal?

**Comment by nicoddemus** [comment]
I think Travis is backed-up... I opened #1767, #1768, and #1769 yesterday and they have not ran on Travis yet... 😞

**Comment by goanpeca** [comment]
fucked up you mean 😆 ?

**Comment by nicoddemus** [comment]
Heh! I saw this before in the last package I contributed as well.

I'm surprised, my impression of Travis is that it starts up really quickly. It might be that conda-forge is triggering a ton of builds, or might be due to it building OSX packages... I don't really know. 😅

**Comment by jakirkham** [comment]
So in situations like this, please check our [status page]( https://conda-forge.github.io/status ) and/or [Travis CI's status page]( https://www.traviscistatus.com/ ) plus any other relevant status pages based on the issue. In this case, it is [known issue]( https://www.traviscistatus.com/incidents/b4xk2bybjc9d ) that Travis CI has been working on over the last day. Progress is likely to be slow throughout the day.

**Comment by goanpeca** [comment]
Yeah, I noticed, well there goes my new awesome package release that depended on this one ;-).

Next week it is, its travel weekend for me!

**Comment by nicoddemus** [comment]
Oh good to know it's an anomaly. 😄

**Comment by jakirkham** [comment]
Sorry for the inconvenience guys. :disappointed: Thanks for your understanding.

Have a safe trip, @goanpeca .

**Comment by nicoddemus** [comment]
> Sorry for the inconvenience guys. 

Please not at all! Thanks to you and the entire conda-forge team for the brilliant environment! 😁

**Comment by goanpeca** [comment]
Thanks :-) no need to appologize ;-)

**Comment by havocp** [comment]
appveyor failure looks like it could be a transient network problem that would be fixed by clicking "rebuild commit" if anyone has that permission

**Comment by nicoddemus** [comment]
@havocp just did it, thanks for the ping!

File: recipes/pytest-json/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    - pytest >=2.3
+    - python
+
+test:
+  imports:
+    - pytest_json
+
+about:
+  home: https://github.com/mattcl/pytest-json
+  license: MIT License
```
Comment (jakirkham) [inline recipes/pytest-json/meta.yaml]: Please shorten to `MIT`. Also do they have a license file we can include?

File: recipes/pytest-json/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  summary: Generate JSON test reports
+  description: |
+    pytest-json is a plugin for py.test that generates JSON reports for test
+    results.
+
+extra:
+  recipe-maintainers:
+    - goanpeca
```
Comment (jakirkham) [inline recipes/pytest-json/meta.yaml]: nit: Please keep a terminal newline.

**Comment by jakirkham** [comment]
Basically fine. Some trivial stuff left.

File: recipes/pytest-json/meta.yaml (hunk 3 above)
Comment (jakirkham) [inline recipes/pytest-json/meta.yaml]: Also may optionally add `license_family: MIT`.

**Comment by goanpeca** [comment]
@jakirkham merge :-p ?

**Comment by jakirkham** [comment]
:+1: Will do once it passes.

File: recipes/pytest-json/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  {{ hash_type }}: {{ hash }}
+
+build:
+    script: python setup.py install --single-version-externally-managed --record=record.txt
+    number: 0
+
+requirements:
+  build:
```
Comment (jakirkham) [inline recipes/pytest-json/meta.yaml]: Oops. 😳 Missing `python`.

File: recipes/pytest-json/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  {{ hash_type }}: {{ hash }}
+
+build:
+    script: python setup.py install --single-version-externally-managed --record=record.txt
+    number: 0
```
Comment (jakirkham) [inline recipes/pytest-json/meta.yaml]: Normally build `number` comes before `script`. Though it's not strictly required to.

**Comment by jakirkham** [comment]
Sorry. 😞 Missed some things when I was looking at it earlier. Have updated them. All should be easy though.

**Comment by goanpeca** [comment]
No biggie, pushed the fixes

File: recipes/pytest-json/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    - python
+
+test:
+  imports:
+    - pytest_json
+
+about:
+  home: https://github.com/mattcl/pytest-json
+  license: MIT
+  license_family: MIT
```
Comment (goanpeca) [inline recipes/pytest-json/meta.yaml]: https://github.com/mattcl/pytest-json/issues/1

**Comment by jakirkham** [comment]
Thanks @goanpeca .

**Pull request merged**